libs/core/src/lib/select/select-mobile/select-mobile/select-mobile.component.ts

```diff
@@ -1,41 +1,67 @@
-import { AfterViewInit, Component, ElementRef, OnDestroy, OnInit, TemplateRef, ViewChild } from '@angular/core';
+import {
+    AfterViewInit,
+    Component,
+    ElementRef,
+    Inject,
+    Injector,
+    OnDestroy,
+    OnInit,
+    Optional,
+    TemplateRef,
+    ViewChild
+} from '@angular/core';
 import { DialogRef } from '../../../dialog/dialog-utils/dialog-ref.class';
 import { DialogService } from '../../../dialog/dialog-service/dialog.service';
-import { SelectComponent } from '../../select.component';
 import { Subscription } from 'rxjs';
 import { MobileModeConfig } from '../../../utils/interfaces/mobile-mode-config';
-import { OptionComponent } from '../../option/option.component';
+import { FD_OPTION_PARENT_COMPONENT, FdOptionParentComponent } from '../../option/option.component';
 
+/**
+ * This component provides extended mobile support for Select component to render list of option since full screen
+ * dialog.
+ */
 @Component({
     selector: 'fd-select-mobile',
     templateUrl: './select-mobile.component.html'
 })
 export class SelectMobileComponent implements OnInit, AfterViewInit, OnDestroy {
-
-    /** @hidden Dialog template reference */
-    @ViewChild('dialogTemplate') dialogTemplate: TemplateRef<any>;
+    /**
+     * Dialog template reference
+     *
+     * @hidden
+     */
+    @ViewChild('dialogTemplate')
+    dialogTemplate: TemplateRef<any>;
 
     /** @hidden */
     dialogRef: DialogRef;
 
     /** @hidden */
     childContent: TemplateRef<any> = undefined;
 
-    /** @hidden */
-    private _selectedOnOpen: OptionComponent;
-
     /** @hidden */
     private _subscriptions = new Subscription();
 
+    private _activeItemIndex: number;
+
     constructor(
         private _elementRef: ElementRef,
         private _dialogService: DialogService,
-        private _selectComponent: SelectComponent
-    ) { }
+        private _injector: Injector,
+        @Optional() @Inject(FD_OPTION_PARENT_COMPONENT) private _parent: FdOptionParentComponent
```

Comment:
- Why is this `@Optional`? I don't think SelectMobile view should be ever used without parent dependency. Also implementation of SelectMobile component doesn't expect `_parent` to be missing.
- It looks like `private _injector: Injector` is also not needed.